akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -269,7 +269,7 @@ final class GroupBy[T, K](maxSubstreams: Int, keyFor: T ⇒ K) extends GraphStag
               subSubstreamSource.push(nextElementValue)
               clearNextElement()
             }
-          } else tryPull(in)
+          } else if (!hasBeenPulled(in)) tryPull(in)
```

Comment:
I would deprecate the current one (but not remove) and replace it with `canBePulled` or something more sane. Also replace `isAvailable` with `hasElementPending`. The names are ofc just working names, my point is to make it less surprising and more in-line with the output port (`isAvailable` -> `canBePushed`). Something like that.